test/bdd/exec_test.go

```diff
@@ -0,0 +1,326 @@
+
+import (
+	"flag"
+	"os"
+	"os/exec"
+	"regexp"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	scheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+
+	"github.com/litmuschaos/chaos-operator/pkg/apis/litmuschaos/v1alpha1"
+	chaosClient "github.com/litmuschaos/chaos-operator/pkg/client/clientset/versioned/typed/litmuschaos/v1alpha1"
+	restclient "k8s.io/client-go/rest"
+)
+
+var (
+	kubeconfig      string
+	config          *restclient.Config
+	k8sClientSet    *kubernetes.Clientset
+	litmusClientSet *chaosClient.LitmuschaosV1alpha1Client
+)
+
+func init() {
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "path to kubeconfig to invoke kubernetes API calls")
+}
+func TestChaos(t *testing.T) {
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "BDD test")
+}
+
+var _ = BeforeSuite(func() {
+
+	var err error
+	kubeconfig = os.Getenv("HOME") + "/.kube/config"
+	config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get config")
+	}
+
+	k8sClientSet, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get k8sClientSet")
+	}
+
+	litmusClientSet, err = chaosClient.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get litmusClientSet")
+	}
+
+	err = v1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		klog.Infof("Error to add to Scheme: %v", err)
+	}
+
+	//Creating crds
+	By("Installing Litmus CRDs")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/chaos_crds.yaml").Run()
+	if err != nil {
+		klog.Infof("Unable to execute command, due to error: %v", err)
+	}
+
+	//Creating rbacs
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/rbac.yaml").Run()
+	if err != nil {
+		klog.Infof("Unable to execute command, due to error: %v", err)
+	}
+
+	//Creating Chaos-Operator
+	By("Installing Chaos-Operator")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/operator.yaml").Run()
+	if err != nil {
+		klog.Infof("Unable to execute command, due to error: %v", err)
+	}
+
+	klog.Infof("Chaos-Operator installed Successfully")
+
+	//Wait for the creation of chaos-operator
+	time.Sleep(10 * time.Second)
+
+	//Check for the status of the chaos-operator
+	operator, _ := k8sClientSet.CoreV1().Pods("litmus").List(metav1.ListOptions{LabelSelector: "name=chaos-operator"})
+	for _, v := range operator.Items {
+
+		Expect(string(v.Status.Phase)).To(Equal("Running"))
+		break
+	}
+})
+
+//BDD Tests to check secondary resources
+var _ = Describe("BDD on chaos-executor", func() {
+
+	var err error
+	// BDD TEST CASE 1
+	When("Create a test Deployment with nginx image", func() {
+		It("Should create Nginx deployment ", func() {
+			By("building a deployment")
```

Comment:
```suggestion
			By("building an nginx deployment object")
```

Reply:
I dont this object is really needed, will edit to make it more understandable